src/ducks/reimbursements/HealthReimbursements.jsx

```diff
@@ -3,24 +3,42 @@ import { connect } from 'react-redux'
 import { queryConnect } from 'cozy-client'
 import { transactionsConn } from 'doctypes'
 import { flowRight as compose, sumBy, groupBy } from 'lodash'
+import flag from 'cozy-flags'
 import { getHealthExpensesByPeriod } from 'ducks/filters'
 import { TransactionsWithSelection } from 'ducks/transactions/Transactions'
 import withBrands from 'ducks/brandDictionary/withBrands'
 import withAppsUrls from 'ducks/apps/withAppsUrls'
-import { isFullyReimbursed } from 'ducks/transactions/helpers'
+import {
+  isFullyReimbursed,
+  getReimbursementStatus
+} from 'ducks/transactions/helpers'
 import { translate } from 'cozy-ui/react'
 import { Title } from 'cozy-ui/react/Text'
 import { Padded } from 'components/Spacing'
 import { Figure } from 'components/Figure'
 import styles from 'ducks/reimbursements/HealthReimbursements.styl'
 
 class DumbHealthReimbursements extends Component {
+  getGroups() {
+    return groupBy(this.props.filteredTransactions, getReimbursementStatus)
+  }
+
   render() {
     const { filteredTransactions, t } = this.props
-    const {
-      true: reimbursedTransactions,
-      false: awaitingTransactions
-    } = groupBy(filteredTransactions, isFullyReimbursed)
+    const reimbursementTagFlag = flag('reimbursement-tag')
+
+    const groupedTransactions = groupBy(
+      filteredTransactions,
+      reimbursementTagFlag ? getReimbursementStatus : isFullyReimbursed
+    )
+
+    const reimbursedTransactions = reimbursementTagFlag
+      ? groupedTransactions.reimbursed
+      : groupedTransactions.true
+
+    const awaitingTransactions = reimbursementTagFlag
+      ? groupedTransactions.pending
+      : groupedTransactions.false
```

Comment:
I am not a fan of doing groupBys in the render. Would it be easy to hide that behind a selector (easily memoizable if we need it).

Reply:
Unfortunately I can't do this now, so I added a comment above the grouping code to say that it should be extracted to a selector. Sorry